### DISCUSSION TITLE
When the Windows client (10) connects to the StrongSwan server, the server's internet IP is lost

### DISCUSSION THREAD

**Discussion by TelDragon** [strongswan/strongswan#2200] [Q&A] [answered]
My server configuration

The internet IP of the server is 222.222.222.222 (demo)

```
connections {
    swanctl_Server1_EAP {
        version = 2
        proposals = default,3des-modp1024-prfsha1-sha1,aes128-sha256-x25519
        unique = never
        pools = Private_Network_1
        send_certreq = yes
        send_cert = always
        local {
            auth = pubkey
            certs = IPSecServerCert.pem
            id = ipsec.xxx.xx.cn
        }
        remote {
            auth = eap-mschapv2
            eap_id = %any
        }
        children {
            swanctl_Server1_EAP {
                start_action = start
                dpd_action = restart
                local_ts  = 172.17.62.0/24,192.168.0.0/24,192.168.1.0/24
                remote_ts = 0.0.0.0/0
                updown = /usr/local/strongswan/libexec/ipsec/_updown iptables
                esp_proposals = default,3des-modp1024-prfsha1-sha1,aes128-sha256-x25519
            }
        }
    }
}

pools {
    Private_Network_1 {
        addrs = 51.51.51.0/24
    }
}

secrets {

   eap-ct_Cloud {
      id = ct_Cloud
      secret = ******
   }
}
```

I have two Linux clients in two regions that can connect normally and use their connection pools.
Can ping the private network address and internet of the communication server

But once I connect to Windows.
The server's internet is like losing it. Others are unable to ping this IP properly.

When the Windows client is not connected. Anyone, including my Linux client, can ping normally 222.222.222.222

After Windows client authentication and login, no one, including my Linux client, can ping 222.222.222.222
I feel like it has nothing to do with my Windows client routing（Whether executed or not）
```
Remove-VpnConnectionRoute -ConnectionName "ipsec.xxx.xx.cn" -DestinationPrefix 172.17.62.0/24 -PassThru


DestinationPrefix : 172.17.62.0/24
InterfaceIndex    :
InterfaceAlias    : ipsec.xxx.xx.cn
AddressFamily     : IPv4
NextHop           : 0.0.0.0
Publish           : 0
RouteMetric       : 1
PolicyStore       :

```

The problem may occur on the traffic selector of the server configuration

I am very confused.

**Comment by tobiasbrunner** [comment]
> remote_ts = 0.0.0.0/0

Don't configure this when assigning virtual IPs from a pool, just keep it to the default (`dynamic`).

**Reply by TelDragon** [reply]
```
children {
            swanctl_Server1_EAP {
                start_action = start
                dpd_action = restart
                local_ts  = 172.17.62.0/24,192.168.0.0/24,192.168.1.0/24
                remote_ts = dynamic
                updown = /usr/local/strongswan/libexec/ipsec/_updown iptables
                esp_proposals = default,3des-modp1024-prfsha1-sha1,aes128-sha256-x25519
            }
```

Indeed, the problem was localized to the remote traffic selector.
But after I changed the parameters, my Linux client couldn't connect properly.

```
 <swanctl_Server1_EAP|6> traffic selectors 172.17.62.0/24 192.168.1.0/24 === 192.168.0.0/24 unacceptable
Mon, 2024-04-08, 21:36:56 10[IKE1] <swanctl_Server1_EAP|6> failed to establish CHILD_SA, keeping IKE_SA
Mon, 2024-04-08, 21:36:56 10[ENC1] <swanctl_Server1_EAP|6> generating IKE_AUTH response 5 [ AUTH CPRP(ADDR) N(MOBIKE_SUP) N(ADD_4_ADDR) N(TS_UNACCEPT) ]
```

This is my previously connected Linux client traffic selector. Can't the remote client also be written as dynamic by default?
```
remote_ts = 172.17.62.0/24,192.168.0.0/24
local_ts = 192.168.1.0/24
```

**Reply by tobiasbrunner** [reply]
> Can't the remote client also be written as dynamic by default?

Yes, `local_ts = dynamic` is the recommended setting for clients that request a virtual IP from the server.

**Reply by TelDragon** [reply]
I will provide my modification information as follows

linux client1    192.168.1.5
```
remote_ts = 172.17.62.0/24,192.168.0.0/24
local_ts = dynamic
```

linux client2     192.168.0.188

```
remote_ts = 172.17.62.0/24,192.168.1.0/24
local_ts = dynamic
```

windows10 client  192.168.1.31

```
add-VpnConnectionRoute -ConnectionName "ipsec.xxx.xx.cn" -DestinationPrefix 172.17.62.0/24 -PassThru
```

I found that I have learned another pattern.
In this mode, the local encryption IP used in the client is not a private IP. But it's a virtual IP

**Reply by tobiasbrunner** [reply]
> In this mode, the local encryption IP used in the client is not a private IP. But it's a virtual IP

That's not what you want? Then why request/offer a virtual IP to the clients?

**Reply by TelDragon** [reply]
Indeed, this situation is not the result I wanted,
Because I found that the server is unable to access the client's private IP address....
The server can currently only access the client's virtual IP
```
swanctl_Server1_EAP: #4, ESTABLISHED, IKEv2, d3c1667402299089_i fc0db7b2a017319c_r*
  local  'ipsec.xxx.xx.cn' @ 172.17.62.205[4500]
  remote '192.168.0.188' @ 14.xxx[4500] EAP: 'xx1_Cloud' [51.51.51.2]
  AES_CBC-128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/ECP_256
  established 143s ago, rekeying in 13431s
  swanctl_Server1_EAP: #2, reqid 2, INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-128/HMAC_SHA2_256_128
    installed 143s ago, rekeying in 3244s, expires in 3817s
    in  c2a1fecd,   1092 bytes,    13 packets,    93s ago
    out c06b9610,    336 bytes,     4 packets,   117s ago
    local  172.17.62.0/24 192.168.1.0/24
    remote 51.51.51.2/32
swanctl_Server1_EAP: #3, ESTABLISHED, IKEv2, 381f3d7f8de2f6dd_i eacf425c3d99007a_r*
  local  'ipsec.xxx.xx.cn' @ 172.17.62.205[4500]
  remote '192.168.1.5' @ 101.xxxx[9480] EAP: '51EPD_Hebei' [51.51.51.1]
  AES_CBC-128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/ECP_256
  established 213s ago, rekeying in 13019s
  swanctl_Server1_EAP: #1, reqid 1, INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-128/HMAC_SHA2_256_128
    installed 213s ago, rekeying in 3255s, expires in 3747s
    in  c6b47ab3,  37908 bytes,   499 packets,     0s ago
    out c012e302,  49177 bytes,   435 packets,     5s ago
    local  172.17.62.0/24 192.168.0.0/24
    remote 51.51.51.1/32
```

Because before, my server could still access the client's private IP addresses, such as 192.168.1.5 and 192.168.0.188

**Reply by tobiasbrunner** [reply]
That's the whole point of virtual IPs, i.e. that you don't have to worry about such potentially conflicting private IPs (and the routing to them). If you want something else, change the config. But as see with Windows clients, that might not work as expected (if you configure `remote_ts=0.0.0.0/0` on the server, clients can redirect whatever traffic they want to themselves).

**Reply by TelDragon** [reply]
Here is my original configuration

Server ip  172.17.62.205
```
local_ts  = 172.17.62.0/24,192.168.0.0/24,192.168.1.0/24
remote_ts = 0.0.0.0/0
```

linux client1 ip 192.168.1.5
```
remote_ts = 172.17.62.0/24,192.168.0.0/24
local_ts = 192.168.1.0/0
```

linux client2 ip 192.168.0.188
```
remote_ts = 172.17.62.0/24,192.168.1.0/24
local_ts = 192.168.0.0/0
```
Accessible on the server  192.168.1.5/192.168.1.1.......  、192.168.0.188.....

Accessible on Linux client 1  172.17.62.205/172.17.62.203........ 、192.168.0.188......

Accessible on Linux client 2  172.17.62.205/172.17.62.203........ 、192.168.1.5/192.168.1.1......

Because there are other business systems under my Linux client that require reverse connection

But I'm a bit confused about the Windows client and Android client. The current configuration is not applicable to these two clients. After they connect, the remote connection pool will always be 0.0.0.0/0

Is this a problem with the Windows client itself?
Or do I need to enable another service configuration item for my special and complex needs? For example, “swanctl_Server2_EAP”

**Reply by tobiasbrunner** [reply]
> But I'm a bit confused about the Windows client and Android client. The current configuration is not applicable to these two clients. After they connect, the remote connection pool will always be 0.0.0.0/0

That's because they expect a virtual IP, so they propose 0.0.0.0/0 as their traffic selector (they don't know what virtual IP will be assigned, so they accept any IP). And again, if configure the same as remote TS on the server, then that's what will be accepted.


> Or do I need to enable another service configuration item for my special and complex needs? For example, “swanctl_Server2_EAP”

You probably want to configure two separate configs, one for clients that request and use a virtual IP and one for clients that don't use a virtual IP (i.e. for site-to-site setups).

**Reply by TelDragon** [reply]
Yes, your thinking is right.
And I also enabled the second configuration file. Distinguish between two methods.
The first method involves two different connectors under the same configuration file

```
connections {
    linux_EAP {
        version = 2
        proposals = default,3des-modp1024-prfsha1-sha1,aes128-sha256-x25519
        unique = never
        pools = Private_Network_1
        send_certreq = yes
        send_cert = always
        local {
            auth = pubkey
            certs = IPSecServerCert.pem
            id = ipsec.xxx.xx.cn
        }
        remote {
            auth = eap-mschapv2
            eap_id = %any
        }
        children {
            linux_EAP {
                start_action = start
                dpd_action = restart
                local_ts  = 172.17.62.0/24,192.168.0.0/24,192.168.1.0/24
                remote_ts = 0.0.0.0/0
                updown = /usr/local/strongswan/libexec/ipsec/_updown iptables
                esp_proposals = default,3des-modp1024-prfsha1-sha1,aes128-sha256-x25519
            }
        }
    }
 windows_android_EAP {
        version = 2
        proposals = default,3des-modp1024-prfsha1-sha1,aes128-sha256-x25519
        unique = never
        pools = Private_Network_1
        send_certreq = yes
        send_cert = always
        local {
            auth = pubkey
            certs = IPSecServerCert.pem
            id = ipsec.xxx.xx.cn
        }
        remote {
            auth = eap-mschapv2
            eap_id = %any
        }
        children {
            windows_android_EAP {
                start_action = start
                dpd_action = restart
                local_ts  = 172.17.62.0/24,192.168.0.0/24,192.168.1.0/24
                remote_ts = dynamic
                updown = /usr/local/strongswan/libexec/ipsec/_updown iptables
                esp_proposals = default,3des-modp1024-prfsha1-sha1,aes128-sha256-x25519
            }
        }
    }
}

pools {
    Private_Network_1 {
        addrs = 51.51.51.0/24
    }
}

secrets {

   eap-ct_Cloud {
      id = ct_Cloud
      secret = ******
   }
}
```
In this way, all clients still adapt to linux-EAP and it does not work

I am foolish. The second way I experiment is to generate two independent configuration files (. conf), but the content is the same.
linux_EAP.conf   
windows_android_EAP.conf

But they ultimately all pointed to the first connector. Namely linux-EAP

I feel that the options applicable to Windows and Android special clients depend on
Pools=PrivateNetwork_1 parameter and remote_ts=dynamic

**Reply by tobiasbrunner** [reply]
Configs can only be differentiated based on IPs and identities.

**Reply by TelDragon** [reply]
Do you have a better and simpler method? Please advise me. I don't want to use the client certificate method at the moment because EAP is relatively simple and universal besides PSK.

**Reply by tobiasbrunner** [reply]
As I said, you have to use different IPs or identities to identify which config to use.

**Reply by TelDragon** [reply]
Is it similar to distinguishing by ID identity?

```
secrets {

   eap-ct_Cloud1 {
      id = ct_Cloud1
      secret = ******
   }
   eap-ct_Cloud2 {
      id = ct_Cloud2
      secret = ******
   }
}
```

the server1 conf
```
remote {
            auth = eap-mschapv2
            eap_id = ct_Cloud1
        }
```

the server2 conf
```
remote {
            auth = eap-mschapv2
            eap_id = ct_Cloud2
        }
```

But it seems that it cannot be connected....

**Reply by tobiasbrunner** [reply]
No, EAP-Identity != IKE Identity. And you can't match on the EAP-Identity without a workaround (see e.g. #1326).

**Reply by TelDragon** [reply]
Sorry. I got confused.

[https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html)

```
default

id = %any

IKE identity to expect for authentication round. Refer to the id keyword of the [local](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html#_connections_conn_local) section for details. When using certificate-based authentication, the IKE identity must be contained in the certificate, either as the subject DN or as a subjectAltName. It’s possible to use wildcards to match remote identities (e.g. *@strongswan.org, *.strongswan.org or "C=CH, O=strongSwan, CN=*"). Connections with exact matches are preferred. When using DNs with wildcards, the charon.rdn_matching option in [strongswan.conf](https://docs.strongswan.org/docs/5.9/config/strongswanConf.html) specifies how Relative Distinguished Names (RDNs) are matched

meanwhile

eap_id = id

Identity to use as peer identity during EAP authentication. If set to %any the EAP-Identity method will be used to ask the client for an EAP identity

```

I might be doing this

```
remote {
            auth = eap-mschapv2
            eap_id = %any
            id = ct_Cloud1
        }

```

**Reply by TelDragon** [reply]
How can the same connection match authentication with multiple IDs? I only allow a certain number of IDs to be connected to a specific license, not just one ID.

Can it be like this?

```
remote {
            auth = eap-mschapv2
            eap_id = %any
            id = ct_Cloud1,ct_Cloud2,ct_Cloud3
        }
```

Can it be like this?

```
remote {
            auth = eap-mschapv2
            eap_id = %any
            id = ct_Cloud1 ct_Cloud2 ct_Cloud3
        }
```

**Reply by tobiasbrunner** [reply, marked as answer]
You can only match multiple identities via wildcards (e.g. `*@strongswan.org`). And it will only work if clients actually send useful IKE identities (e.g. Windows doesn't if EAP authentication is used, it just sends the private IP address). Matching the server identity has a similar issue as some clients don't send a remote identity (again, Windows is one of them, strongSwan's Android client as well, by default). But either might be OK, if you want to use one of the connections only for very specific clients (that can send a proper local or remote IKE identity), and the other connection (that hasn't configured any specific local or remote identity) is the fallback for all other clients. And as mentioned above, you could match on the EAP-Identity, you just have to use the workaround via dummy connection.

**Reply by TelDragon** [reply]
Indeed, I have also noticed this issue. Windows clients do not send valid ID identifiers. Its identifier is just a combination of the client's IP address and the user. EAP: 'ct-Cloud2' [51.51.51.3]

This forces me to use multiple connections to implement it
